tests/bluetooth/host/keys/bt_keys_foreach_bond/src/test_suite_foreach_bond_invalid_inputs.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <bluetooth/bluetooth.h>
+#include "mocks/keys_help_utils.h"
+
+/* Callback to be used when no calls are expected by bt_foreach_bond() */
+static void bt_foreach_bond_unreachable_cb(const struct bt_bond_info *info, void *user_data)
+{
+	zassert_unreachable("Unexpected call to '%s()' occurred", __func__);
+}
+
+static void test_case_setup(void *f)
+{
+	clear_key_pool();
+}
+
+ZTEST_SUITE(bt_keys_foreach_bond_invalid_inputs, NULL, NULL, test_case_setup, NULL, NULL);
+
+/*
+ *  Test callback function is set to NULL
+ *
+ *  Constraints:
+ *   - Any ID value can be used
+ *   - Callback function pointer is set to NULL
+ *
+ *  Expected behaviour:
+ *   - An assertion fails and execution stops
+ */
+ZTEST_EXPECT_FAIL(bt_keys_foreach_bond_invalid_inputs, test_null_callback);
+ZTEST(bt_keys_foreach_bond_invalid_inputs, test_null_callback)
+{
+	bt_foreach_bond(0x00, NULL, NULL);
+}
+
+/*
+ *  Test if the callback will be called if the ID doesn't exist
+ *
+ *  Constraints:
+ *   - ID doesn't exist in the list
+ *   - Valid callback is passed to bt_keys_foreach_bond()
+ *
+ *  Expected behaviour:
+ *   - Callback should never be called
+ */
+ZTEST(bt_keys_foreach_bond_invalid_inputs, test_callback_non_existing_id)
```

Comment:
can you add another one checking no callbacks are called when the keys pool/list is empty (but the other params are valid)?